base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -1991,6 +2035,88 @@ impl fmt::Display for MetadataValue {
             MetadataValue::PrunedHeight(height) => write!(f, "Effective pruned height is {}", height),
             MetadataValue::BestBlock(hash) => write!(f, "Chain tip block hash is {}", hash.to_hex()),
             MetadataValue::HorizonData(_) => write!(f, "Horizon data"),
+            MetadataValue::DeletedBitmap(deleted) => {
+                write!(f, "Deleted Bitmap ({} indexes)", deleted.bitmap().cardinality())
+            },
+        }
+    }
+}
+
+/// A struct that wraps a LMDB transaction and provides an interface to valid operations that can be performed
+/// on the current deleted bitmap state of the blockchain.
+/// A deleted bitmap contains the MMR leaf indexes of spent UTXOs.
+struct BlockchainDeletedBitmap<'a, T> {
+    txn: &'a T,
+    db: &'a Database<'static>,
+    cache: Option<DeletedBitmap>,
+    is_dirty: bool,
+}
+
+impl<'a, 'b, T> BlockchainDeletedBitmap<'a, T>
+where T: Deref<Target = ConstTransaction<'b>>
+{
+    pub fn new(txn: &'a T, db: &'a Database<'static>) -> Self {
+        Self {
+            txn,
+            db,
+            cache: None,
+            is_dirty: false,
+        }
+    }
+
+    /// Returns a reference to the `DeletedBitmap`, loading the bitmap if required.
+    pub fn get(&mut self) -> Result<&DeletedBitmap, ChainStorageError> {
```

Comment:
Strange that this method requires a mut, perhaps a better name or comment explaining why the mut is necessary

Reply:
I've changed the code to not require get to be mutable